mobilecoind/src/payments.rs

```diff
@@ -179,17 +216,17 @@ impl<T: BlockchainConnection + UserTxConnection + 'static, FPR: FogPubkeyResolve
     ) -> Result<(u64, u32), Error> {
         // Figure out the block_version and fee, taking into account if opt_fee is
         // nonzero
-        let candidate_block_version = self.ledger_db.get_latest_block()?.version;
+        let block_version = max(
+            self.ledger_db.get_latest_block()?.version,
+            last_block_info.network_block_version,
+        );
+
         Ok(if opt_fee != 0 {
-            (opt_fee, candidate_block_version)
+            (opt_fee, block_version)
```

Comment:
i think it was slightly buggy that we were not considering `last_block_info.network_block_version` here before